processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -194,16 +207,81 @@ protected void scanAndAggregate(
   )
   {
     final Cursor cursor = params.getCursor();
-    if (specializeGeneric1AggPooledTopN && theAggregators.length == 1) {
-      scanAndAggregateGeneric1Agg(params, positions, theAggregators[0], cursor);
-    } else if (specializeGeneric2AggPooledTopN && theAggregators.length == 2) {
+    if (theAggregators.length == 1) {
```

Comment:
Can we abstract the predicate / function checking into a `List` of `Pair`s  of predicates and runnables or similar? and loop through the list until it finds a matching predicate to run? Then to deny `specializeHistoricalSingleValueDimSelector1SimpleDoubleAggPooledTopN` is simply to not add it in the list.